include/gz/sim/MeshInertiaCalculator.hh

```diff
@@ -0,0 +1,230 @@
+ *
+ */
+
+#ifndef GZ_SIM_MESHINERTIACALCULATOR_HH_
+#define GZ_SIM_MESHINERTIACALCULATOR_HH_
+
+#include <optional>
+
+#include <sdf/sdf.hh>
+#include <sdf/InterafaceMoiCalculator.hh>
+
+#include "gz/sim/Util.hh"
+
+#include <gz/common/graphics.hh>
+#include <gz/common/Mesh.hh>
+#include <gz/common/MeshManager.hh>
+
+#include <gz/math/Vector3.hh>
+#include <gz/math/MassMatrix3.hh>
+
+
+namespace gz
+{
+namespace sim
+{
+inline namespace GZ_SIM_VERSION_NAMESPACE
+{
+  struct Triangle
+  {
+    gz::math::Vector3d v0, v1, v2;
+  };
+  
+  class GZ_SIM_VISIBLE MeshInertiaCalculator
+  {
+    public: void GetMeshTriangles(std::vector<Triangle>& triangles,
+            const gz::common::Mesh* mesh);
+
+    public: void CalculateMassProperties(std::vector<Triangle>& triangles, 
+            const double density, gz::math::MassMatrix3d& massMatrix);
+
+    public: std::optional<gz::math::MassMatrix3d> CalculateMeshInertia(sdf::Errors& _errors, 
+            const sdf::InterfaceMoiCalculator& _calculatorParams);
+
+    public: std::optional<gz::math::MassMatrix3d> operator()(sdf::Errors& _errors,
+            const sdf::InterfaceMoiCalculator& _calculatorParams);
+    
+    //public: sdf::CustomMOICalculator customCalculator = CalculateMeshInertia;
+  };
+
+  void MeshInertiaCalculator::GetMeshTriangles(std::vector<Triangle>& triangles,
+          const gz::common::Mesh* mesh)
+  {
+    double* vertArray = NULL;
+    int* indArray = NULL;
+
+    mesh->FillArrays(&vertArray, &indArray);
+
+    // Write code to check if size of the ind array is divisible by 3
+
+    for (unsigned int i = 0; i < mesh->IndexCount(); i += 3)
+    {
+      Triangle triangle;
+      triangle.v0.X() = vertArray[3 * indArray[i]];
+      triangle.v0.Y() = vertArray[3 * indArray[i] + 1];
+      triangle.v0.Z() = vertArray[3 * indArray[i] + 2];
+
+      triangle.v1.X() = vertArray[3 * indArray[i+1]];
+      triangle.v1.Y() = vertArray[3 * indArray[i+1] + 1];
+      triangle.v1.Z() = vertArray[3 * indArray[i+1] + 2];
+
+      triangle.v2.X() = vertArray[3 * indArray[i+2]];
+      triangle.v2.Y() = vertArray[3 * indArray[i+2] + 1];
+      triangle.v2.Z() = vertArray[3 * indArray[i+2] + 2];
+
+      triangles.push_back(triangle);
+    }
+    std::cout << "Number of Indices of the mesh: " << mesh->IndexCount() << std::endl;
+    std::cout << "Number of Vertices of the mesh: " << mesh->VertexCount() << std::endl;
+  }
+
+  void MeshInertiaCalculator::CalculateMassProperties(std::vector<Triangle>& triangles,
+            const double density, gz::math::MassMatrix3d& massMatrix)
+  {
+    const double coefficients[10] = {1.0 / 6, 1.0 / 24, 1.0 / 24, 1.0 / 24, 1.0 / 60,
+                                       1.0 / 60, 1.0 / 60, 1.0 / 120, 1.0 / 120, 1.0 / 120};
+
+    // Number of triangles. Would probably add this as a data member of the MeshAdapterClass
+    int numTriangles = triangles.size();
+    std::cout << "Number of Traingles of the mesh: " << numTriangles << std::endl;
```

Comment:
use `gzdbg` or remove

Reply:
Removed in [c79a397](https://github.com/gazebosim/gz-sim/pull/2061/commits/c79a397140370dff5c9fa65aa73d7ce321b4cd9b)